test/jdk/javax/management/remote/nonLocalAccess/NonLocalJMXRemoteTest.java

```diff
@@ -63,40 +75,100 @@
+                    + "-J-Dcom.sun.management.jmxremote.ssl=false \\\n"
+                    + "-J-Dcom.sun.management.jmxremote.authenticate=false \\\n"
+                    + "-J-Dcom.sun.management.jmxremote.host=<registry-host>"
+                    + "\n\nRegistry service is run in the background without any "
+                    + "output. Enter the hostname or IP address of the different "
+                    + "host and the port separated by a semicolon below and continue "
+                    + "the test."
+                    + "\n\n",
+    };
+    static final String message = "Enter <registry.host>:<port> and submit:";
+    static final int TIMEOUT_MS = 3600000;
+    private volatile boolean abort = false;
+
     public static void main(String[] args) throws Exception {
 
-        String host = System.getProperty("jmx-registry.host");
-        if (host == null || host.isEmpty()) {
-            throw new RuntimeException("Specify host with system property: -Djmx-registry.host=<host>");
-        }
-        int port = Integer.getInteger("jmx-registry.port", -1);
-        if (port <= 0) {
-            throw new RuntimeException("Specify port with system property: -Djmx-registry.port=<port>");
+        String host = System.getProperty("registry.host");
+        int port = Integer.getInteger("registry.port", -1);
+        if (host == null || host.isEmpty() || port <= 0) {
+            NonLocalJMXRemoteTest test = new NonLocalJMXRemoteTest();
+
+            int testId = 0;
+            if (args.length > 0) {
+                testId = Integer.valueOf(args[0]);
+            }
+            String input = Utils.readHostInput(
+                    "NonLocalJMXRemoteTest",
+                    instructions[testId],
+                    message,
+                    TIMEOUT_MS
+            );
+            String[] hostAndPort = input.split(":");
+
+            if (hostAndPort.length >= 1) {
+                host = hostAndPort[0];
+                port = 8888;
+            }
+
+            if (hostAndPort.length == 2) {
+                port = Integer.valueOf(hostAndPort[1]);
+            }
+
+            if (host == null || host.isEmpty() || port <= 0) {
+                throw new RuntimeException(
+                        "Specify host with system property: -Dregistry"
+                                + ".host=<host> and -Dregistry.port=<port>");
+            }
         }
 
         // Check if running the test on a local system; it only applies to remote
         String myHostName = InetAddress.getLocalHost().getHostName();
-        Set<InetAddress> myAddrs = Set.of(InetAddress.getAllByName(myHostName));
-        Set<InetAddress> hostAddrs = Set.of(InetAddress.getAllByName(host));
+        Set<InetAddress> myAddrs =
+                Set.copyOf(Arrays.asList(InetAddress.getAllByName(myHostName)));
+        Set<InetAddress> hostAddrs =
+                Set.copyOf(Arrays.asList(InetAddress.getAllByName(host)));
```

Comment:
Is there a reason to change this from Set.of?

Reply:
This is a workaround for duplicate elements detected by Set.of(). On my localhost, InetAddress.getAllByName(localhost returns duplicate IPv6 addresses with different scoped id which is not counted during comparison. This change shouldn't impact the goal of these tests which is to verify that registry modification request from client side is prohibited.

Reply:
Thanks. You might want to add a comment to avoid drive-by suggestions to simplify it.